### ISSUE TITLE
[BUG] Calendar selects previous date in certain scenario

### ISSUE THREAD

**Issue by RasmusKjeldgaard** [kirbydesign/designsystem#2147]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue)  

## Describe the bug
When `timezone="UTC"` is enabled on the calendar, and you are in a timezone that is _before_ London (GMT _minus_ something), and you click a date, the calendar changes to the previous month, with that date selected instead.

## Describe how to reproduce the bug

1. Go to [calendar cookbook page](https://cookbook.kirby.design/#/home/showcase/calendar)
2. Enable the 'Use UTC instead of local time' in the calendar
3. Change the timezone used in the browser (For example by enabling the 'Sensors' tool in Chrome Dev tools), and set your location to e.g. San Francisco, or any other location to the west of london. 
4. Refresh
5. See that the selected date jumps back one day on click.

## Which Kirby version was used?
6.2.2

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/stable/.github/SUPPORT.md#team-kirby); do not proceed until this is done

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [stable branch](https://github.com/kirbydesign/designsystem/tree/stable) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

**Comment by RasmusKjeldgaard** [comment]
Closing this issue automatically because it has been open for more than 6 months and is not currently in progress. If this is still relevant, please feel free to reopen it or open a new issue with updated details. Thank you!
